mlflow/pipelines/steps/train.py

```diff
@@ -125,88 +123,129 @@ def _run(self, output_directory):
             MLFLOW_PIPELINE_PROFILE_NAME: self.step_config["profile"],
             MLFLOW_PIPELINE_STEP_NAME: run_args.get("step", ""),
         }
-        with mlflow.start_run(tags=tags) as run:
-            estimator.fit(X_train, y_train)
-
-            if hasattr(estimator, "best_score_"):
-                mlflow.log_metric("best_cv_score", estimator.best_score_)
-            if hasattr(estimator, "best_params_"):
-                mlflow.log_params(estimator.best_params_)
-
-            # TODO: log this as a pyfunc model
-            code_paths = [os.path.join(self.pipeline_root, "steps")]
-            estimator_schema = infer_signature(X_train, estimator.predict(X_train.copy()))
-            logged_estimator = mlflow.sklearn.log_model(
-                estimator,
-                f"{self.name}/estimator",
-                signature=estimator_schema,
-                code_paths=code_paths,
-            )
-            # Create a pipeline consisting of the transformer+model for test data evaluation
-            with open(transformer_path, "rb") as f:
-                transformer = cloudpickle.load(f)
-            mlflow.sklearn.log_model(transformer, "transform/transformer", code_paths=code_paths)
-            model = make_pipeline(transformer, estimator)
-            model_schema = infer_signature(raw_X_train, model.predict(raw_X_train.copy()))
-            model_info = mlflow.sklearn.log_model(
-                model, f"{self.name}/model", signature=model_schema, code_paths=code_paths
+
+        if self.step_config["tuning_enabled"]:
+            # gate all HP tuning code within this condition
+            tuning_params = self.step_config["tuning"]
+            tuning_method = self.step_config["using"]  # pylint: disable=unused-variable
```

Comment:
Not quite following this line - variable is unused. It seems like we should check if `self.step_config["using"] == estimator_spec` and, if it is, throw because we can't have tuning enabled and `using != estimator_spec`.

Reply:
good point. the handling of `step_config["using"]` happens in the `from_pipeline_config()` function already, so removing here.